packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -2127,71 +2090,44 @@ class _ScrollUnderFlexibleSpace extends StatelessWidget {
         break;
     }
 
-    late final Widget? collapsedTitle;
     late final Widget? expandedTitle;
     if (title != null) {
-      collapsedTitle = config.collapsedTextStyle != null
-        ? DefaultTextStyle(
-            style: config.collapsedTextStyle!.copyWith(color: foregroundColor ?? appBarTheme.foregroundColor),
-            child: title!,
-          )
-        : title;
+      final TextStyle? expandedTextStyle = titleTextStyle
+        ?? appBarTheme.titleTextStyle
+        ?? config.expandedTextStyle?.copyWith(color: foregroundColor ?? appBarTheme.foregroundColor);
       expandedTitle = config.expandedTextStyle != null
         ? DefaultTextStyle(
-            style: config.expandedTextStyle!.copyWith(color: foregroundColor ?? appBarTheme.foregroundColor),
+            style: expandedTextStyle!,
             child: title!,
           )
         : title;
     }
 
-    late final bool centerTitle;
-    {
-      bool platformCenter() {
-        switch (theme.platform) {
-          case TargetPlatform.android:
-          case TargetPlatform.fuchsia:
-          case TargetPlatform.linux:
-          case TargetPlatform.windows:
-            return false;
-          case TargetPlatform.iOS:
-          case TargetPlatform.macOS:
-            return true;
-        }
+    EdgeInsetsGeometry expandedTitlePadding() {
+      final EdgeInsets padding = config.expandedTitlePadding!.resolve(Directionality.of(context));
+      if (bottomHeight > 0) {
+        return EdgeInsets.fromLTRB(padding.left, 0, padding.right, bottomHeight);
       }
-      centerTitle = centerCollapsedTitle ?? appBarTheme.centerTitle ?? platformCenter();
-    }
-
-    EdgeInsetsGeometry effectiveCollapsedTitlePadding = EdgeInsets.zero;
-    if (hasLeading && leadingWidth == null) {
-      effectiveCollapsedTitlePadding = centerTitle
-        ? config.collapsedCenteredTitlePadding!
-        : config.collapsedTitlePadding!;
-    } else if (hasLeading && leadingWidth != null) {
-      effectiveCollapsedTitlePadding = EdgeInsetsDirectional.only(start: leadingWidth!);
+      if (theme.useMaterial3) {
+        final TextStyle textStyle = config.expandedTextStyle!;
+        // Substract the bottom line height from the bottom padding.
+        // TODO(tahatesser): Figure out why this is done.
+        // https://github.com/flutter/flutter/issues/120672
+        final double adjustBottomPadding = padding.bottom
+          - (textStyle.fontSize! * textStyle.height! - textStyle.fontSize!) / 2;
```

Comment:
Expanded title bottom padding fix for https://github.com/flutter/flutter/issues/104493 uses M3 default text style. More details [here ](https://github.com/flutter/flutter/issues/120672#issuecomment-1463747486).

`textStyle.height!` is null in M2 TextTheme` and this bottom padding adjustment is done only for M3.

Reply:
Hm this is a really good point. The medium and large variants were added for M3. Do they look like M3 by default mostly? What do they look like in M2? I don't know that we ned to answer these questions in the context of this PR (it is already doing a bit too much for one PR), but maybe we should file a bug and make sure these variants look and work nicely in M2 and M3.

Reply:
>  Let me know if you want to duplicate it, I can move the original copy of this test into M2 group of tests.

I spoke too soon, I actually already added defaults in M2 Group and also added title offset checks.  
So Both M2 and M3 default tests are part of this PR. Hence M3 explicit M3 test is here.

> the medium and large variants were added for M3. Do they look like M3 by default mostly? What do they look like in M2?

I did some investigation. The padding and title offset looks good, with no cut-off in M2 so no bottom padding adjustments aren't needed in M2. 

However foreground color was incorrect so I just fixed this.

![Screenshot 2023-03-23 at 11 06 06](https://user-images.githubusercontent.com/48603081/227182038-0a1b13c4-e835-49a7-871f-57da36752c57.png)